impl/src/main/java/org/glassfish/json/JsonParserImpl.java

```diff
@@ -55,25 +56,29 @@
 public class JsonParserImpl implements JsonParser {
 
     private final BufferPool bufferPool;
+    private final boolean allowDuplicateKeys;
     private Context currentContext = new NoneContext();
     private Event currentEvent;
 
     private final Stack stack = new Stack();
     private final JsonTokenizer tokenizer;
 
-    public JsonParserImpl(Reader reader, BufferPool bufferPool) {
+    public JsonParserImpl(Reader reader, BufferPool bufferPool, boolean allowDuplicateKeys) {
```

Comment:
I'm not big fan of this approach - are we going to add new argument to the constructor for every property in the future? Having `Map<String,?> options` here instead would probably look better to me

Reply:
I was following the existing code path / design. I have cleaned things up a bit. Please see my latest commit.